pyi.py

```diff
@@ -417,14 +421,27 @@ def visit_Assign(self, node: ast.Assign) -> None:
                     target_info = TypeVarInfo(cls_name=cls_name, name=target_name)
                     self.typevarlike_defs[target_info] = node
                 else:
-                    self.error(target, Y001.format(cls_name))
+                    self.error(node, Y001.format(cls_name))
+
+        Y015_error = False
         if isinstance(node.value, (ast.Num, ast.Str, ast.Bytes)):
             self._Y015_error(node)
+            Y015_error = True
+        elif isinstance(node.value, (ast.Constant, ast.NameConstant)):
+            if (
+                not isinstance(node.value, ast.Ellipsis)
+                and node.value.value is not None
+            ):
+                self._Y015_error(node)
+                Y015_error = True
+
         # We avoid triggering Y026 for calls and = ... because there are various
         # unusual cases where assignment to the result of a call is legitimate
         # in stubs.
-        elif target_name != "__all__" and not isinstance(
-            node.value, (ast.Ellipsis, ast.Call)
+        if (
+            not Y015_error
+            and target_name != "__all__"
+            and not isinstance(node.value, (ast.Ellipsis, ast.Call))
```

Comment:
It's only this part of the diff that's relevant to fixing the bug. The rest of the diff was necessary because flake8 complained that we'd exceeded our complexity limit, so I had to factor out the Y017 checks into a separate function.